aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -0,0 +1,247 @@
+        } else {
+            stake_info.current_stake = stake_info.current_stake + TestCoin::value(&delegation.coin);
+            Vector::push_back(&mut stake_info.active, delegation);
+        }
+    }
+
+    /// Withdraw from active delegation, it's moved to cooldown if locked_until_secs < current_time or
+    /// directly deposit if it's not from an active validator.
+    fun withdraw_active(account: &signer, from: address) acquires ValidatorInfo, ValidatorSet {
+        let addr = Signer::address_of(account);
+        let current_time = Timestamp::now_seconds();
+        let stake_info = &mut borrow_global_mut<ValidatorInfo>(from).stake_info;
+        let d = withdraw_internal(&mut stake_info.inactive, addr);
+        let is_validator = is_validator(from);
+        if (!is_validator) {
+            // directly deposit if it's not active validator
+            let Delegation {coin, from: _, locked_until_secs: _} = d;
+            TestCoin::deposit(addr, coin);
+        } else if (d.locked_until_secs < current_time) {
+            // move to cooldown if it can be unlocked
+            Vector::push_back(&mut stake_info.cooldown, d);
+        } else {
+            // not allowed to withdraw
+            abort 0
+        };
+    }
+
+    /// Withdraw from inactive delegation, directly deposited to the account's balance.
+    fun withdraw_inactive(account: &signer, from: address) acquires  ValidatorInfo {
+        let addr = Signer::address_of(account);
+        let stake_info = &mut borrow_global_mut<ValidatorInfo>(from).stake_info;
+        let d = withdraw_internal(&mut stake_info.inactive, addr);
+        let Delegation {coin, from: _, locked_until_secs: _} = d;
+        TestCoin::deposit(addr, coin);
+    }
+    /// Initialize the ValidatorInfo for account.
+    fun register_validator_candidate(account: &signer, consensus_pubkey: vector<u8>, network_address: vector<u8>) {
+        let stake_info = StakeInfo {
+            current_stake: 0,
+            active: Vector::empty(),
+            warmup: Vector::empty(),
+            cooldown: Vector::empty(),
+            inactive: Vector::empty(),
+        };
+        move_to(account, ValidatorInfo {
+            stake_info,
+            consensus_pubkey,
+            network_address,
+        });
+    }
+
+    /// Rotate the consensus key of the validator.
+    fun rotate_consensus_key(account: &signer, consensus_pubkey: vector<u8>) acquires ValidatorInfo, ValidatorSet {
+        let addr = Signer::address_of(account);
+        let validator_set = borrow_global_mut<ValidatorSet>(@CoreResources);
+        if (Vector::contains(&validator_set.validators, &addr)) {
+            on_new_epoch(validator_set);
+        };
+        borrow_global_mut<ValidatorInfo>(addr).consensus_pubkey = consensus_pubkey;
+    }
+
+    /// Initialize validator set to the core resource account.
+    fun initialize_validator_set(account: &signer, minimum_stake: u64, maximum_stake: u64) {
```

Comment:
what to do if we want to raise the minium_stake after we operating the aptos chain for a while?

Reply:
it's probably the same as all the privileged operations that need a voting mechanism